src/libertem/api.py

```diff
@@ -95,8 +96,12 @@ def load(self, filetype: str, *args, **kwargs) -> DataSet:
 
     load.__doc__ = load.__doc__ % {"types": ", ".join(filetypes.keys())}
 
-    def create_mask_job(self, factories, dataset, use_sparse=None,
-                        mask_count=None, mask_dtype=None, dtype=None) -> ApplyMasksJob:
+    MaskArrayType = Union[np.ndarray, sp.coo.coo_matrix, sp.dok.dok_matrix]
+    
+    def create_mask_job(self,
+            factories: Union[Callable[[], MaskArrayType], Iterable[Callable[[], MaskArrayType]]],
```

Comment:
The type for `factories` could also be extracted as an alias - as you are re-using the same construct in the `create_mask_analysis` function below. This new type could also be extracted to a specific module, like `libertem.masks`.